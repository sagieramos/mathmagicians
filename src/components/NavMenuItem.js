import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavMenuItem = ({ to, children }) => (
  <li>
    <NavLink to={to}>
      {children}
    </NavLink>
  </li>
);

NavMenuItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default NavMenuItem;
