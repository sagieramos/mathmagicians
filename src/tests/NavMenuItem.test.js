// Import necessary dependencies
import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import NavMenuItem from '../components/NavMenuItem';

test('renders NavMenuItem component correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <NavMenuItem to="/calculator">Example</NavMenuItem>
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});
