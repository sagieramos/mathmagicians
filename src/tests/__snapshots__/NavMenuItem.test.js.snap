// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavMenuItem component correctly 1`] = `
<div>
  <li>
    <a
      class=""
      href="/calculator"
    >
      Example
    </a>
  </li>
</div>
`;
